.changeset/slow-tools-pull.md

```diff
@@ -0,0 +1,5 @@
+---
+'@leafygreen-ui/modal': major
+---
+
+Move className prop to apply to root `div` rather than overlay `div`, and add contentClassName to style overlay container
```

Comment:
Might want to note the direct upgrade path for consumers (e.g. change instances of `className` to `contentClassName`)